core/domain/user_services.py

```diff
@@ -138,7 +169,35 @@ def get_user_settings_from_email(email):
         return get_user_settings(user_model.id)
 
 
-def get_users_settings(user_ids, strict=False, include_marked_deleted=False):
+@overload
+def get_users_settings(
+    user_ids: Sequence[Optional[str]],
+    strict: Literal[True] = ...,
+    include_marked_deleted: bool = False
+) -> Sequence[user_domain.UserSettings]: ...
+
+
+@overload
+def get_users_settings(
+    user_ids: Sequence[Optional[str]],
+    strict: Literal[False] = ...,
+    include_marked_deleted: bool = False
+) -> Sequence[Optional[user_domain.UserSettings]]: ...
+
+
+@overload
+def get_users_settings(
+    user_ids: Sequence[Optional[str]],
+    strict: bool = False,
+    include_marked_deleted: bool = False
+) -> Sequence[Optional[user_domain.UserSettings]]: ...
+
+
+def get_users_settings(
+    user_ids: Sequence[Optional[str]],
+    strict: bool = False,
+    include_marked_deleted: bool = False
+) -> Sequence[Optional[user_domain.UserSettings]]:
```

Comment:
for the implementation, the return type is `List[Optional[user_domain.UserSettings]]`.

But if `strict=True` provided, the return type is `List[user_domain.UserSettings]` and for `strict=false` the return type is `List[Optional[user_domain.UserSettings]]`. 

Also, from Invariant property of List, we know `List[aaa]` cannot be a subtype of `List[Optional[aaa]]`. But `List[aaa]` can be subtype of `Sequence[Optional[aaa]]`. So that's why i used `Sequence` here.